cmd/ingester/app/consumer/consumer.go

```diff
@@ -103,6 +106,9 @@ func (c *Consumer) handleMessages(pc sc.PartitionConsumer) {
 		c.logger.Debug("Got msg", zap.Any("msg", msg))
 		msgMetrics.counter.Inc(1)
 		msgMetrics.offsetGauge.Update(msg.Offset)
+		if !msg.BlockTimestamp.IsZero() { // This timestamp is zero for Kafka broker (or the broker data format) < v0.10.2
+			msgMetrics.lagGaugeNs.Update(c.timeSince(msg.BlockTimestamp).Nanoseconds())
```

Comment:
Why gauge? Out metrics API supports timers natively, and probably accept `time.Duration`.

Reply:
@yurishkuro, @black-adder bought this up earlier to which I had replied

> I'm concerned about the value of having those accurate metrics because offset lag is computed in an approximate way - the block offset from Kafka is not set per message but per message block and is dependent on Kafka broker timestamps. Having timers might give a greater sense of accuracy compared to gauges.

Reply:
Call it batch lag, I think it addresses the ambiguity much better than using guage instead of timer.